example/src/main/java/com/klarna/example/MainActivity.java

```diff
@@ -19,23 +20,69 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setContentView(R.layout.activity_main);
 
-        Context.setApiKey("test_d8324b98-97ce-4974-88de-eaab2fdf4f14");
-    }
+        com.klarna.ondemand.Context.setApiKey("test_d8324b98-97ce-4974-88de-eaab2fdf4f14");
 
-    public void openKlarnaRegistration(View view) {
-        Intent intent = new Intent(this, RegistrationActivity.class);
-        startActivityForResult(intent, KLARNA_REGISTRATION_REQUEST_CODE);
+        initializeUIElements();
     }
 
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == KLARNA_REGISTRATION_REQUEST_CODE) {
+        if (requestCode == REGISTRATION_REQUEST_CODE) {
             switch (resultCode) {
-                case WebViewActivity.RESULT_CANCELED:
+                case RegistrationActivity.RESULT_OK:
+                    String token = data.getStringExtra(RegistrationActivity.EXTRA_USER_TOKEN);
+                    saveUserToken(token);
+                    findViewById(R.id.registerButton).setVisibility(View.INVISIBLE);
+                    findViewById(R.id.preferencesButton).setVisibility(View.VISIBLE);
+                    break;
+                case RegistrationActivity.RESULT_CANCELED:
+                case RegistrationActivity.RESULT_ERROR:
```

Comment:
In the iOS demo we had a comment along the lines of 
```
// You may also want to convey this failure to your user.
```
when things fail. We should add one here as well.